libs/ui-lib/lib/common/components/clusterWizard/clusterDetailsValidation.ts

```diff
@@ -119,20 +120,20 @@ export const getClusterDetailsValidationSchema = ({
         ? baseDomainValidationSchema.required('Required')
         : dnsNameValidationSchema.required('Required'),
       pullSecret: pullSecretValidationSchema.required('Required.'),
-      SNODisclaimer: Yup.boolean().when(['highAvailabilityMode', 'openshiftVersion'], {
+      SNODisclaimer: Yup.boolean().when(['controlPlaneCount', 'openshiftVersion'], {
         // The disclaimer is required only if SNO is enabled and SNO feature is not fully supported in that version
         is: (
-          highAvailabilityMode: Cluster['highAvailabilityMode'],
+          controlPlaneCount: Cluster['controlPlaneCount'],
           openshiftVersion: Cluster['openshiftVersion'],
         ) => {
           const selectedVersion = (ocpVersions || []).find((v) => v.value === openshiftVersion);
+          const versionToUse = selectedVersion?.value ?? openshiftVersion;
           if (featureSupportLevels) {
-            return highAvailabilityMode === 'None' && selectedVersion && featureSupportLevels
-              ? featureSupportLevels.getFeatureSupportLevel(selectedVersion.value, 'SNO') ===
-                  'dev-preview'
+            return toNumber(controlPlaneCount) === 1 && versionToUse && featureSupportLevels
+              ? featureSupportLevels.getFeatureSupportLevel(versionToUse, 'SNO') === 'dev-preview'
               : false;
```

Comment:
Isn't the ternary operator completely unnecessary? 

```suggestion
            return toNumber(controlPlaneCount) === 1 && versionToUse && featureSupportLevels
              && featureSupportLevels.getFeatureSupportLevel(versionToUse, 'SNO') === 'dev-preview';
```

Reply:
i'll take a look